fleetspeak/src/server/servertests/comms_test.go

```diff
@@ -263,6 +264,115 @@ func TestBlacklist(t *testing.T) {
 	}
 }
 
+// blocklistService is a Fleetspeak service.Service that counts blocklisted
+// and non-blocklisted messages.
+type blocklistService struct {
+	blocklistedCount    *uint
+	nonBlocklistedCount *uint
+}
+
+func (s blocklistService) Start(sctx service.Context) error { return nil }
+func (s blocklistService) ProcessMessage(ctx context.Context, m *fspb.Message) error {
+	if m.IsBlocklistedSource {
+		(*s.blocklistedCount)++
+	} else {
+		(*s.nonBlocklistedCount)++
+	}
+	return nil
+}
+func (s blocklistService) Stop() error { return nil }
+
+func TestStoredMessagesFromBlocklistedClient(t *testing.T) {
+	fin := sertesting.SetServerRetryTime(func(_ uint32) time.Time {
+		return db.Now().Add(time.Second)
+	})
+	defer fin()
+
+	ctx := context.Background()
+	var blocklistedCount uint = 0
+	var nonBlocklistedCount uint = 0
+	testService := blocklistService{&blocklistedCount, &nonBlocklistedCount}
+
+	ts := testserver.MakeWithService(t, "server", "Blocklist", testService)
+	defer ts.S.Stop()
+
+	k, err := ts.AddClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	id, err := common.MakeClientID(k)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Blacklist the client
+	if err := ts.DS.BlacklistClient(ctx, id); err != nil {
+		t.Fatalf("BlacklistClient returned error: %v", err)
+	}
+
+	// Put a message in the database that would otherwise be ready for delivery.
+	mid, err := common.RandomMessageID()
+	if err != nil {
+		t.Fatalf("Unable to create message id: %v", err)
+	}
+	clientMessage := &fspb.Message{
+		MessageId:       mid.Bytes(),
+		SourceMessageId: []byte("AAABBBCCC"),
+		Source: &fspb.Address{
+			ServiceName: "TestService",
+			ClientId:    id.Bytes(),
+		},
+		Destination: &fspb.Address{
+			ServiceName: "TestService",
+		},
+		MessageType:  "TestMessage",
+		CreationTime: db.NowProto(),
+	}
+
+	if err := ts.DS.StoreMessages(ctx, []*fspb.Message{clientMessage}, ""); err != nil {
+		t.Fatalf("Unable to store message: %v", err)
+	}
+
+	tctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	for {
+		msgs, err := ts.DS.GetMessages(tctx, []common.MessageID{mid}, true)
+		if err != nil {
+			t.Logf("GetMessages failed: %v", err)
+			goto Skip
+		}
+		if len(msgs) != 1 {
+			t.Fatalf("Expected 1 message, got: %v", msgs)
```

Comment:
Here and all the messages below: please identify the function, the input, got/want.

Example: 
`t.Fatalf("GetMessages(%s) returned %d messages, want 1", mID, len(msgs))`

https://google.github.io/styleguide/go/decisions#identify-the-function
https://google.github.io/styleguide/go/decisions#identify-the-input
https://google.github.io/styleguide/go/decisions#got-before-want

Reply:
Done.